### DISCUSSION TITLE
It´s possible to run two mastodon instances in the same Debian 11 server?

### DISCUSSION THREAD

**Discussion by CiroArtigot** [mastodon/mastodon#22105] [Q&A]
Hello,
Does anyone know if you can run two instances of Mastodon on the same server?
I use Debian 11 and I would like my server to have two instances of Mastodon running for two different domains.
Thanks

**Comment by abdessalaam** [comment]
I am thinking of this too (on Ubuntu). I have found [this article](https://nfgworld.com/multiple-mastodon-instances-on-a-single-server/). I am also wondering about the server resources needed...? [see 22062](https://github.com/mastodon/mastodon/discussions/22062)

**Reply by CiroArtigot** [reply]
Hello!

I think that the article is for have a second Mastodon instance (not two instances at the same time).

Probably the problem are the listenig ports, 3000 for backend and 4000 for streaming.
When you install a second Mastodon instance, it try to listen to the same ports.
I'm looking to modifiy the dist/*. services to listen 3001 and 4001, but it doesnt´t work. i think there is a problem with this nginx configuration:

upstream backend {
    server 127.0.0.1:3001 fail_timeout=0;
}

upstream streaming {
    server 127.0.0.1:4001 fail_timeout=0;
}

Do i need to change "backend" and "streaming" names? if not, the nginx gives a duplicate upstream error..
this probably need more code changes..

I don´t know  what happens when you install mastodon in a server with ports 3000/4000 listening to another app...

Of course i'm not an expert, but culd be a good idea to have an option to confiure listenig ports for each instance?

Thanks!!!
Ciro

**Comment by wiegelmann** [comment]
Also you need to separate Redis by using a namespace, and another Postgres db, and another Elastic index. Running in Docker could be a better option for separation because it is more isolated on the network level.

**Reply by CiroArtigot** [reply]
Thanks, i will try to do that in Docker!
